### ISSUE TITLE
`@time` reports greater than 100% compilation time on v1.9.0-beta2

### ISSUE THREAD

**Issue by jishnub** [JuliaLang/julia#48155]
This is on v1.9.0-beta2, so the fix in https://github.com/JuliaLang/julia/pull/41286 doesn't seem to apply to this.
In a fresh session, I run
```julia
julia> @time @eval sin.(zeros(10, 10));
  0.200751 seconds (280.38 k allocations: 18.425 MiB, 130.97% compilation time)

julia> versioninfo()
Julia Version 1.9.0-beta2
Commit 7daffeecb8c (2022-12-29 07:45 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 8 × 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-14.0.6 (ORCJIT, tigerlake)
  Threads: 1 on 8 virtual cores
Environment:
  LD_LIBRARY_PATH = :/usr/lib/x86_64-linux-gnu/gtk-3.0/modules
  JULIA_EDITOR = subl
```
On v1.8.4, I obtain
```julia
julia> @time @eval sin.(zeros(10, 10));
  0.183622 seconds (593.69 k allocations: 30.280 MiB, 95.88% compilation time)
```
so this seems like a regression.

**Comment by giordano** [comment]
Probably duplicate of #48024, fixed by  #48033, it'd be useful to test on master

**Comment by jishnub** [comment]
Indeed, seems fixed on nightly. I'll close this, as this should be fixed on beta3.
